File: homebot.py
#!/usr/bin/python
# Sets and imports everything needed to make the program run
import time
import os
import sys
import config
allgood=False
allgood2=False
allgood3=False
meal=""
freq=""
famlike=""
ingravailable=""

with open("instructions.txt") as f:
        lines = f.readlines()

lines = [line.rstrip('\n') for line in open("instructions.txt")]
# Simple little script to enable scroll typing
def sprint(s):
    for c in s:
        sys.stdout.write(c)
        sys.stdout.flush()
        time.sleep(0.02)
os.system("clear")
sprint("Booting HomeBotOS...")
time.sleep(2)
os.system("clear")

def main():
    # Asks user baseline questions
    def questions():
        os.system("clear") # Clears the console screen
        sprint("Alright %s, what do you want tonight? " % name) # Why isn't this workingahdujlfdhjsfhkdslfjsdfhslkahfkjh8ewuifhdskj
        sprint("We have a few things on the menu currently.\nType the number that corresponds with the food you want!\n")
        time.sleep(1) # Pauses for one second
        sprint("\n\nMENU:\n\n1. Chicken\n2. Kraft dinner\n3. Potatoes\n")
        global meal
        menu = input()
        if menu == "1":
            meal = "chicken"
        if menu == "2":
            meal = "kraft dinner"
        if menu == "3":
            meal = "potatoes"
        sprint("Awesome! I'll help you make %s." % (meal))
        time.sleep(1)
        sprint("\n\nHow many times do you want me to help you make %s this week? (Answer with a number.)\n" % meal)
        timeweek=input()
        time.sleep(1)
        sprint("\nOk, I'll help you make %s %s times this week." % (meal, timeweek))
        time.sleep(2)
    # Tells the user what ingredients they need
    def ingredients():
        global ingravailable
        global ingredientcheck
        ingredientcheck = False
        while not ingredientcheck: # When the user says "y" or "n", ingredientcheck is equal to True, otherwise it repeats the function
            os.system("clear")
            if meal == 'chicken':
                sprint(config.ingredients["chicken"] + " (y/n)") # Pulls in ingredients from the config.py file and appends (y/n) to the end of it
                ingravailable=input()
            elif meal == 'kraft dinner':
                sprint(config.ingredients["kd"] + " (y/n)")
                ingravailable=input()
            elif meal == 'potatoes':
                sprint(config.ingredients["potatoes"] + " (y/n)")
                ingravailable=input()

        # Tells the user whether they're good to go or not
            if ingravailable == 'y':
                sprint("Great. You're all set!")
                time.sleep(2)
                ingredientcheck = True
            elif ingravailable == 'n':
                sprint("All right. If you go get those ingredients, I'll help you when you get back. Deal?")
                ingredientcheck = True
                sys.exit()
            else:
                sprint("Whoops! Something went wrong! Try again.")
                time.sleep(2)
                ingredientcheck = False
    # Gives the user instructions on how to cook their food
    def instructions():
        os.system("clear")
        if meal == "chicken":
            for i in config.chicken: # Iterates over the line numbers in the config.py file
                sprint(lines[i]) # Prints one instruction line at a time
                print("\n\nPress [Enter] to continue.\n")
                input()
                os.system("clear")
        if meal == "kraft dinner":
            for i in config.kd:
                print(lines[i])
                sprint("\nPress [Enter] to continue.\n")
                input()
                os.system("clear")
        if meal == "potatoes":
            for i in config.potatoes:
                print(lines[i])
                sprint("\nPress [Enter] to continue.\n")
                input()
                os.system("clear")

    def introduction():
        sprint("Hello! I am HomeBot, your personal cooking helper.\nI am going to help you with all your cooking needs! ")
        time.sleep(3)

        sprint("\n\nFirst of all, what's your name?\n\n")
        global name
        name = input()
        print("variable" + name)
        f = open('config.py', 'a')
        f.write("\nname = dict(\n    yourname = \"" + name + "\",\n)")
# Introduction
    try:
        if config.name["yourname"]: # if there is a "name" module
            global name
            name = config.name["yourname"]
    except:
            introduction() # Make this include all below stuff

    questions()
    ingredients()
    instructions()

if __name__ == "__main__":
    main()
    print("Main")

File: homebot.c
#include <stdio.h> //standard input/output; for slow_print and scanf
#include <stdbool.h> //standard bool; for booleans
#include <stdlib.h> //find out what this is for
#include <unistd.h> //for sleep/delay functions
#include <string.h> //for slow_print fucntion

void slow_print (char *strprint);
void slow_print_wait (char *strprint);
void intro();
void chicken(void);
void kraftdinner(void);
void potatoes(void);

char ingredients = 'a';

int if_selection = 0, menu_selection = 0; //declares and initializes two variables


int fgetc(FILE *stream);

int main(void) //main function
{
	int if_selection = 0, menu_selection = 0; //declares and initializes two variables
	bool program_running = true;
    char name[20];
	int config = 0;

	FILE *settings;
	settings = fopen("settings.txt", "r");

	config = fgetc(settings);

	fclose(settings);

	system("clear");

	if (config == 'a')
	{
	intro();
	FILE *settings;
	settings = fopen("settings.txt", "w");
	fprintf(settings, "b");
	fclose(settings);
	}

	int fclose(FILE *);

	while (program_running)
	{	
		slow_print("MENU:\n\n1. Chicken\n2. Home-made Kraft Dinner\n3. Potatoes\n\n"); //Prints main menu
		
		scanf("%i", &menu_selection); //Continually scans for an integer typed by the user

		while (if_selection == 0)
		{
			switch (menu_selection)
			{
			case 1: //If the user types 1 (chicken)
				chicken();
				
				break;
				
			case 2: //If user types 2 (kraft dinner)
				kraftdinner();

				break;
				
      		case 3: //If user types 3 (potatoes)
				potatoes();
				
				break;
        
      		default: //If user doesn't type 1, 2, or 3
			system("clear");	
			slow_print("I'm sorry, you didn't enter a correct value. Please try again.\n\n");
			sleep(1);
			system("clear");

			break;
			}
		break;
		} //if_selection while
    } //main While

return 0;
} //main

void slow_print (char *strprint)
{
    int index;
    int string_length = strlen(strprint);
    
    for (index = 0; index < string_length; index++)
    {
        printf("%c", strprint[index]);
        fflush(stdout);
        usleep(20000);
    }
}

void slow_print_wait (char *strprint)
{
    bool instruction_wait = true;
    int index;
    int string_length = strlen(strprint);
    char complete = 'a';

    for (index = 0; index < string_length; index++)
    {
        printf("%c", strprint[index]);
        fflush(stdout);
        usleep(20000);
    }
    
    while (instruction_wait)
    {
        scanf("%c", &complete);
        if (complete == 'y' || complete == 'Y')
        {
            instruction_wait = false;
            system("clear");
        }
        fflush(stdout);
    }

}

void intro(void)
{
	bool program_running = true;
    char name[20];
	
	system("clear");

	slow_print("Hello there! I'm HomeBot, your personal cooking helper! I'll help you with instructions, proportions, and pretty much everything cooking.\n");
	slow_print("So, first I need to ask you some questions so that I can assist you better.");
    sleep(2);
    system("clear");
	slow_print("I never got to ask, what's your name?\n");
	scanf("%s", name);
    slow_print("Hello, "); 
	slow_print(name);
	slow_print("!");

    sleep(1);
	system("clear");
	
}

void chicken(void)
{
	system("clear");
	slow_print("Well, chicken it is!\n\n");
	sleep(1);
	system("clear");

	slow_print("Alright, so if you want some great home-made chicken, you're gonna need these ingredients:\n\n");
	slow_print("1. Whole chicken\n2. Herbs like rosemary, chive, cilantro, etc.\n3. Whole lemon\n4. Pepper and salt\n\n");
	slow_print("\nDo you have all of those?\n\n");
	
	if (getchar() == 89 || getchar() == 121)
	{
		slow_print("\nGreat! You're all set.\n");
		sleep(1);
		system("clear");
	}
	else
	{
		slow_print("Ok, I'll help you make chicken once you get those ingredients.");
		sleep(1);
		system("clear");
		return;
	}

	slow_print_wait("All right. Here\'s the first thing to do: preheat the oven to 400 degrees.\n\n");
	sleep(1);
	slow_print_wait("Rub the chicken all over with salt and pepper. Then, cut slits in the chicken and tuck your herbs in.\n\n");
	sleep(1);
	slow_print_wait("Now, put the diced lemon into the cavity.\n\n");
	sleep(1);
	slow_print_wait("Put the chicken in a roasting pan, and put it in the oven with the cover on. Let it cook for one to one and a half hours, and then lower the temperature to 350 degrees.\n\n");
	sleep(1);
	slow_print_wait("Keep it in until the internal temperature is 165 degrees.\n\n");
	sleep(1);
	slow_print_wait("Once you have it out, you can start to pull it apart. Be careful! It\'s hot! Make sure to pull along the grain, not against it.\n\n");
	sleep(1);
	slow_print_wait("You can now serve it in buns, or eat it by itself.\n\n");

	system("clear");
}

void kraftdinner(void)
{
	system("clear");
	slow_print("Alright, Kraft dinner!\n\n");
	sleep(1);
	system("clear");

	slow_print("Ok, here are the ingredients that you need for kraft dinner:\n\n");
	slow_print("1. One cup elbow noodles\n2. Quarter cup butter\n3. A quarter cup of all-purpose butter\n4. Half a teaspoon of salt\n5. Two cups of milk\n6. 2 Two cups of shredded cheese (cheddar works best)\n7. Pepper and salt\n\n");
	slow_print("Do you have all of those ingredients?\n\n");
	
	if (getchar() == 89 || getchar() == 121)
	{
		slow_print("\nGreat! You're all set.\n");
		sleep(1);
		system("clear");
	}
	else
	{
		slow_print("Ok, I'll help you make chicken once you get those ingredients.");
		sleep(1);
		system("clear");
		return;
	}

	slow_print_wait("First things first; bring a big pot of water to a boil. Cook the elbow noodles in the water (while it's boiling), making sure to stir until cooked but still firm to the bite, usually 8 minutes.\n\n");
	sleep(1);
	slow_print_wait("Drain the noodles now. Melt the butter in a pan over medium heat, and stir the all purpose flour and salt until smooth (usually 5 minutes).\n\n");
	sleep(1);
	slow_print_wait("Slowly pour the milk into the butter flour mixture while continuously stirring until the mix is smooth and bubbling.\n\n");
	sleep(1);
	slow_print_wait("Add shredded cheddar cheese to the milk mixture and stir until the mix is melted, usually 2 to 4 minutes.\n\n");
	sleep(1);
	slow_print_wait("Stir the mixture into the noodles until everything is coated.\n\n");
	sleep(1);
	slow_print_wait("Now you can add a bit more pepper if you want. You can now serve it!\n\n");

	system("clear");
}

void potatoes(void)
{
	system("clear");
	slow_print("Ok, let's have some potatoes!\n\n");
	sleep(1);
	system("clear");
	
	slow_print("First, let's check out the ingredients list to make sure you have everything.\n");
	slow_print("You'll need very little things for this simple meal, but you can always add more if you'd like.\n\n");
	sleep(2);
	system("clear");

	slow_print("You'll need:\n\n1. Potatoes\n2. About two tablespoons of cheese\n3. Salt and pepper\n4. Sour cream\n\n");
	slow_print("Do you have everything?\n\n");

	if (getchar() == 89 || getchar() == 121)
	{
		slow_print("\nGreat! You're all set.\n");
		sleep(1);
		system("clear");
	}
	else
	{
		slow_print("Ok, I'll help you make chicken once you get those ingredients.");
		sleep(1);
		system("clear");
		return;
	}

	slow_print_wait("Make sure to wash and scrub the potato. Prick it in several places with a fork.\n\n");
	sleep(1);
	slow_print_wait("Cook in the microwave for 5 minutes. Then, turn it over and cook for 5 more minutes.\n\n");
	sleep(1);
	slow_print_wait("Season it with as much pepper and salt as you'd like, and mash up the inside a bit with a fork. Top it with some butter and a bit of cheese.\n\n");
	sleep(1);
	slow_print_wait("Return it to the microwave and cook for about one more minute.\n\n");
	sleep(1);
	slow_print_wait("Top it with more cheese, if you'd like, and maybe some sour cream. You can now serve it!\n\n");

	system("clear");
}